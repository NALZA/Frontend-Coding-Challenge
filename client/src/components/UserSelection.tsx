import {FC} from 'react'
import styled from 'styled-components'

const Radio = styled.input`
    -ms-transform: scale(1.5); /* IE 9 */
    -webkit-transform: scale(1.5); /* Chrome, Safari, Opera */
    transform: scale(1.5);
`;

const InputGroup = styled.div`
    display: flex;
    flex-direction: column;
    gap: 20px;
    padding: 20px 0px 20px 0px;
    
`;

const RadioHolder = styled.div`
    display: flex;
    flex-direction: row;
    gap: 10px;
    font-size: 1.17em;
    padding: 15px;

    &.blue{
        background-color: #e4f3ff;
        border-radius: 10px;
    }
`;

interface Props {
    userType: string
    onUserTypeChange: (event: React.ChangeEvent<HTMLInputElement>) => void; 
}

export const UserSelection:FC<Props> = (props) => {
    return (
        <div onChange={props.onUserTypeChange}>
            <h2>User Types</h2>
            <InputGroup>
                <RadioHolder className={props.userType === "Admin" ? "blue": ""}>
                    <Radio type="radio" value="Admin" name="userType" id="admin" data-testid="admin" defaultChecked={props.userType === "Admin"} /> <label htmlFor="admin">Admin</label>
                </RadioHolder>
                <RadioHolder className={props.userType === "Manager" ? "blue": ""}>
                    <Radio  type="radio" value="Manager" name="userType" id="manager" data-testid="manager" defaultChecked={props.userType === "Manager"}/> <label htmlFor="manager">Manager</label>
                </RadioHolder>
            </InputGroup>
        </div>
    )
}
