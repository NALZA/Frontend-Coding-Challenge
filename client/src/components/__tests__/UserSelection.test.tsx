import React from 'react';
import { render } from '@testing-library/react';
import {UserSelection} from '../UserSelection'

describe("renders selection radio buttons", () => {
    it("renders 2 radio buttons with Manager selected", ()=>{
        const handleUserType = (event: React.ChangeEvent<HTMLInputElement>) => {};
        const component = render(<UserSelection userType="Manager" onUserTypeChange={handleUserType} /> );

        const manager = component.getByTestId("manager")
        component.getByTestId("admin")
        expect(manager).toHaveAttribute("checked");
    })
    it("renders 2 radio buttons with Admin selected", ()=>{
        const handleUserType = (event: React.ChangeEvent<HTMLInputElement>) => {};
        const component = render(<UserSelection userType="Admin" onUserTypeChange={handleUserType} /> );

        const admin = component.getByTestId("admin");
        expect(admin).toHaveAttribute("checked");
    })
})