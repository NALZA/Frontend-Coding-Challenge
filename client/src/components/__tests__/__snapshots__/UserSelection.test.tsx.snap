// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders selection radio buttons Match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h2>
          User Types
        </h2>
        <div
          class="sc-gtsrHT eTKwwH"
        >
          <div
            class="sc-dlnjwi kfMQDk blue"
          >
            <input
              checked=""
              class="sc-bdnxRM dgazgZ"
              data-testid="admin"
              id="admin"
              name="userType"
              type="radio"
              value="Admin"
            />
             
            <label
              for="admin"
            >
              Admin
            </label>
          </div>
          <div
            class="sc-dlnjwi kfMQDk"
          >
            <input
              class="sc-bdnxRM dgazgZ"
              data-testid="manager"
              id="manager"
              name="userType"
              type="radio"
              value="Manager"
            />
             
            <label
              for="manager"
            >
              Manager
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h2>
        User Types
      </h2>
      <div
        class="sc-gtsrHT eTKwwH"
      >
        <div
          class="sc-dlnjwi kfMQDk blue"
        >
          <input
            checked=""
            class="sc-bdnxRM dgazgZ"
            data-testid="admin"
            id="admin"
            name="userType"
            type="radio"
            value="Admin"
          />
           
          <label
            for="admin"
          >
            Admin
          </label>
        </div>
        <div
          class="sc-dlnjwi kfMQDk"
        >
          <input
            class="sc-bdnxRM dgazgZ"
            data-testid="manager"
            id="manager"
            name="userType"
            type="radio"
            value="Manager"
          />
           
          <label
            for="manager"
          >
            Manager
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
